nixos/modules/services/networking/harmonia.nix

```diff
@@ -28,6 +28,8 @@ in
   };
 
   config = lib.mkIf cfg.enable {
+    nix.settings.extra-allowed-users = [ "harmonia" ];
```

Comment:
Yeah. This one doesn't break existing config but allow to restrict access.

Reply:
I tested this setting before and only could get it working when not using DynamicUser. It could be that if the nix-daemon starts before harmonia, it doesn't know about this user id.

Reply:
At least in nix it doesn't look like there is any caching done: https://github.com/NixOS/nix/blob/1d86bb4f70ee5c2d06810f21bf7cd057ed46712c/src/nix/daemon.cc#L254

Reply:
also the nixos test seems to work with it: https://github.com/NixOS/nixpkgs/pull/268303/files

Reply:
Not caching in nix itself, but that the namespace doesn't know about the user.

I think it only happened when actually trying to download things and it didn't throw any error and just never found any store entry.

see https://github.com/nix-community/harmonia/issues/106

I had some trouble reproducing it first but I figured it out: The Requires=nix-daemon.socket enforces that the nix-daemon is running before harmonia is running. If I remove that again and first start harmonia and the the nix-daemon I am only greated with missed hash like in the linked issue

Reply:
Interesting. Shouldn't systemd make sure that the socket targets are up before any service? Or is this something that we only encounter in our switch-to-configuration script?

Reply:
Normally yes but in the past I somehow ended up in a situation were that wasn't the case and I got very confused.